modules/java/proxies/java-logmsg-proxy.c

```diff
@@ -49,20 +51,24 @@ Java_org_syslog_1ng_LogMessage_getValue(JNIEnv *env, jobject obj, jlong handle,
 {
   LogMessage *msg = (LogMessage *)handle;
   const gchar *value;
+  gssize value_len;
 
   const char *name_str = (*env)->GetStringUTFChars(env, name, NULL);
   if (name_str == NULL)
     {
       return NULL;
     }
 
-  value = log_msg_get_value_by_name(msg, name_str, NULL);
+  value = log_msg_get_value_by_name(msg, name_str, &value_len);
 
   (*env)->ReleaseStringUTFChars(env, name, name_str);
 
   if (value)
     {
-      return (*env)->NewStringUTF(env, value);
+      gchar stripped_value[value_len + 1];
```

Comment:
I would avoid the allocation of string on the stack, the `value_len` could be big.
You could use the `g_strndup` instead, with that the `\0` termination is guaranteed, although you must `g_free` it after thew `NewStringUTF` call.

Something like that:
```C
gchar *stripped_value = g_strndup(value, value_len);
jstring java_string = (*env)->NewStringUTF(env, stripped_value);
g_free(stripped_value);
return java_string;
```

Note: with this the `#include <string.h>` not needed.

Reply:
strndup() can be slow, and this might be a fastpath, although I am not sure how "fast" it is considering it is called from Java.

Anyway, there's a macro called APPEND_ZERO() which checks if the specific name-value pair is zero terminated already, and avoids the copying in that case. Still, that would allocate memory on the stack, which might be limited to couple of megabytes. but if that's the case, the remaining call sites of APPEND_ZERO would be affected as well.